core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/model/HttpInput.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.alerting.core.model
+
+import org.apache.commons.validator.routines.UrlValidator
+import org.apache.http.client.utils.URIBuilder
+import org.elasticsearch.common.CheckedFunction
+import org.elasticsearch.common.ParseField
+import org.elasticsearch.common.Strings
+import org.elasticsearch.common.unit.TimeValue
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParserUtils
+import java.io.IOException
+
+/**
+ * This is a data class of HTTP type of input for Monitors.
+ */
+data class HttpInput(
+    val scheme: String,
+    val host: String?,
+    val port: Int,
+    val path: String?,
+    val params: Map<String, String>,
+    val url: String,
+    val connection_timeout: Int,
+    val socket_timeout: Int
+) : Input {
+
+    // Verify parameters are valid during creation
+    init {
+        require(connection_timeout > 0) {
+            "Connection timeout: $connection_timeout is not greater than 0."
+        }
+        require(socket_timeout > 0) {
+            "Socket timeout: $socket_timeout is not greater than 0."
+        }
+
+        // Create an UrlValidator that only accepts "http" and "https" as valid scheme and allows local URLs.
+        val urlValidator = UrlValidator(arrayOf("http", "https"), UrlValidator.ALLOW_LOCAL_URLS)
```

Comment:
Seems to me like we shouldn't `ALLOW_LOCAL_URLS`?

Reply:
Need to allow local urls so that monitors can make API calls to local cluster. There are cases of performing local calls in this issue #47 .

Reply:
To monitor the local cluster calls can be made through the LoadBalancer URL (AES) or the hostname (for self-managed clusters).  Allowing localhost access means that the monitor can access any other process listening on the same host.  If we want to allow localhost we need to restrict to 9200 (or whatever the `http.port` setting is).

Reply:
No problem, will add a validation to restrict port to be 9200 when host is equal to localhost.